### DISCUSSION TITLE
Revert Autogenerated Variable Names

### DISCUSSION THREAD

**Discussion by utkonos** [Vector35/binaryninja-api#5206] [Q&A]
I had changed the name of a variable which caused other variables to change to the same name but with `_1` `_2` etc. I later changed the main variable by undefining it (I think; it was a bit ago). I didn't notice that the `_1` variable had not changed back and retained the changed name. I upgraded to 4. Now the undefine symbol choice is not available. I can determine the old name looking at a fresh database built from the sample. But is there a bug here?

<img width="758" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/4df6194d-5555-456b-b862-16a3aa65ebd5">

**Comment by rssor** [comment]
Just to make sure I understand:

1. A variable was given a name
2. That name propagated outwards, with `_1`, `_2`, etc. suffixes
3. The variable naming was undone
4. Variables that the name had propagated to kept those names rather than reverting the initial auto names?

Depending on the exact scenario, that might be expected behavior. Otherwise, is this about names somehow sticking around even if you open/create entirely new databases?

**Comment by utkonos** [comment]
Correct. One aspect of this was the DB was from 3.x and I'm now using it in 4. I can't reproduce any of these issues as much as I have tried after upgrade. The more I think about it. This can probably be closed as just an unfortunate upgrade side effect.
